equinix/helpers_device.go

```diff
@@ -68,64 +69,46 @@ type NetworkInfo struct {
 	PrivateIPv4    string
 }
 
-func getNetworkInfoMetalGo(ips []metalv1.IPAssignment) NetworkInfo {
+func getNetworkInfo(ips []metalv1.IPAssignment) NetworkInfo {
 	ni := NetworkInfo{Networks: make([]map[string]interface{}, 0, 1)}
 	for _, ip := range ips {
 		network := map[string]interface{}{
-			"address": ip.Address,
-			"gateway": ip.Gateway,
-			"family":  ip.AddressFamily,
-			"cidr":    ip.Cidr,
-			"public":  ip.Public,
+			"address": ip.GetAddress(),
```

Comment:
This is replacing direct references to the `ip` struct properties with calls to the `Get...()` functions.  The struct properties are pointers and the `Get...()` functions dereference the pointers (returning the zero value if the pointer is nil), so the `Get...()` functions line up with how the previous `packngo` implementation worked.